File: user-api/src/main/java/hu/aut/bme/userapi/feature/users/controller/UserController.java
package hu.aut.bme.userapi.feature.users.controller;

import hu.aut.bme.userapi.feature.users.dto.CreateUserRequest;
import hu.aut.bme.userapi.feature.users.dto.UpdateUserRequest;
import hu.aut.bme.userapi.feature.users.dto.UserResponse;
import hu.aut.bme.userapi.feature.users.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/users")
class UserController {

    private final UserService userService;

    @GetMapping
    public List<UserResponse> getAllUsers() {
        return userService.findAllUsers();
    }

    @GetMapping("/{userId}")
    public UserResponse getUserById(@PathVariable int userId) {
        return userService.findUserById(userId);
    }

    @PostMapping
    public UserResponse postUser(@RequestBody CreateUserRequest createUserRequest) {
        return userService.createUser(createUserRequest);
    }

    @PutMapping("/{userId}")
    UserResponse updateUser(@PathVariable int userId, @RequestBody UpdateUserRequest updateUserRequest) {
        return userService.updateUser(userId, updateUserRequest);
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @DeleteMapping("/{userId}")
    public void deleteUser(@PathVariable int userId) {
        userService.deleteUserById(userId);
    }
}

File: blog-api/src/main/kotlin/hu/bme/aut/blogapi/feature/users/controller/UserController.kt
package hu.bme.aut.blogapi.feature.users.controller

import hu.bme.aut.blogapi.feature.users.dto.CreateUserRequest
import hu.bme.aut.blogapi.feature.users.dto.UpdateUserRequest
import hu.bme.aut.blogapi.feature.users.dto.UserResponse
import hu.bme.aut.blogapi.feature.users.service.UserService
import org.springframework.data.domain.Sort
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/users")
class UserController(val userService: UserService) {

    @GetMapping
    fun getAllUsersSorted(sort: Sort = Sort.by(Sort.Direction.DESC, "username")): List<UserResponse> {
        return userService.findAllUsersSorted(sort)
    }

    @GetMapping("/{userId}")
    fun getUserById(@PathVariable userId: String): UserResponse {
        return userService.findUserById(userId)
    }

    @PostMapping
    fun postUser(@RequestBody createUserRequest: CreateUserRequest): UserResponse {
        return userService.createUser(createUserRequest)
    }

    @PutMapping("/{userId}")
    fun updateUser(@PathVariable userId: String, @RequestBody updateUserRequest: UpdateUserRequest): UserResponse {
        return userService.updateUser(userId, updateUserRequest)
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @DeleteMapping("/{userId}")
    fun deleteUser(@PathVariable userId: String) {
        userService.deleteUserById(userId)
    }
}